crates/ruma-events/src/secret/request.rs

```diff
@@ -0,0 +1,381 @@
+    CrossSigningSelfSigningKey,
+
+    /// Recovery key (m.megolm_backup.v1).
+    #[ruma_enum(rename = "m.megolm_backup.v1")]
+    RecoveryKey,
+
+    #[doc(hidden)]
+    _Custom(String),
+}
+
+#[cfg(test)]
+mod test {
+    use super::{RequestAction, RequestToDeviceEventContent, SecretName};
+    use matches::assert_matches;
+    use serde_json::{from_value as from_json_value, json, to_value as to_json_value};
+
+    #[test]
+    fn secret_request_serialization() {
+        let content = RequestToDeviceEventContent::new(
+            RequestAction::Request(SecretName::_Custom("org.example.some.secret".into())),
+            "ABCDEFG".into(),
+            "randomly_generated_id_9573".into(),
+        );
+
+        let json = json!({
+            "name": "org.example.some.secret",
+            "action": "request",
+            "requesting_device_id": "ABCDEFG",
+            "request_id": "randomly_generated_id_9573"
+        });
+
+        assert_eq!(to_json_value(&content).unwrap(), json);
+    }
+
+    #[test]
+    fn secret_request_recovery_key_serialization() {
+        let content = RequestToDeviceEventContent::new(
+            RequestAction::Request(SecretName::RecoveryKey),
+            "XYZxyz".into(),
+            "this_is_a_request_id".into(),
+        );
+
+        let json = json!({
+            "name": "m.megolm_backup.v1",
+            "action": "request",
+            "requesting_device_id": "XYZxyz",
+            "request_id": "this_is_a_request_id"
+        });
+
+        assert_eq!(to_json_value(&content).unwrap(), json);
+    }
+
+    #[test]
+    fn secret_request_cancellation_serialization() {
+        let content = RequestToDeviceEventContent::new(
+            RequestAction::RequestCancellation,
+            "ABCDEFG".into(),
+            "randomly_generated_id_9573".into(),
+        );
+
+        let json = json!({
+            "action": "request_cancellation",
+            "requesting_device_id": "ABCDEFG",
+            "request_id": "randomly_generated_id_9573"
+        });
+
+        assert_eq!(to_json_value(&content).unwrap(), json);
+    }
+
+    #[test]
+    fn secret_request_deserialization() {
+        let json = json!({
+            "name": "org.example.some.secret",
+            "action": "request",
+            "requesting_device_id": "ABCDEFG",
+            "request_id": "randomly_generated_id_9573"
+        });
+
+        assert_matches!(
+            from_json_value(json).unwrap(),
+            RequestToDeviceEventContent {
+                action: RequestAction::Request(
+                    SecretName::_Custom(secret)
+                ),
+                requesting_device_id,
+                request_id,
+            }
+            if secret == "org.example.some.secret"
+            && requesting_device_id == "ABCDEFG"
+            && request_id == "randomly_generated_id_9573"
+        )
+    }
+
+    #[test]
+    fn secret_request_cancellation_deserialisation() {
```

Comment:
```suggestion
    fn secret_request_cancellation_deserialization() {
```

for consistency with serde and other parts of the code, please use American English.